src/runtime/hip/hip_hardware_manager.cpp

```diff
@@ -183,6 +183,19 @@ hip_hardware_context::get_property(device_uint_property prop) const {
   case device_uint_property::max_group_size:
     return _properties.maxThreadsPerBlock;
     break;
+  case device_uint_property::max_num_sub_groups:
+    return _properties.maxThreadsPerBlock / _properties.warpSize;
+    break;
+  case device_uint_property::sub_group_independent_forward_progress:
+#ifdef HIPSYCL_RT_HIP_TARGET_CUDA
+    return (_properties.major >= 7) ? 1 : 0; // True on Volta or newer
```

Comment:
I don't think we need to concern ourselves with a potential CUDA path in the HIP backend. There are other places where we assume AMD-specific behavior here. Only a madman would attempt to use hipSYCL's HIP backend to target CUDA instead of just using CUDA ;)